test/table.js

```diff
@@ -1390,6 +1390,238 @@ describe('Bigtable/Table', function() {
     });
   });
 
+  describe('waitForReplication', () => {
+    it('should return the error to the callback', function(done) {
+      var error = new Error('err');
+
+      table.bigtable.request = function(config, callback) {
+        callback(error);
+      };
+
+      table.waitForReplication(function(err) {
+        assert.strictEqual(err, error);
+        done();
+      });
+    });
+
+    it('should call checkConsistency', done => {
+      const consistencyToken = 'sample-token12345';
+
+      table.generateConsistencyToken = function(callback) {
+        callback(null, consistencyToken);
+      };
+
+      table.checkConsistency = function(token, callback) {
+        assert.strictEqual(token, consistencyToken);
+        callback(null, true);
+      };
+
+      table.waitForReplication(() => {
```

Comment:
`table.waitForReplication(done);` is good. It would catch an error should one appear.